### PR TITLE
Cut Cortex 1.4.0 stable

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
Preparation to cut release `1.4.0`.

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]
🎉

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +37,10 @@
    - `-redis.max-conn-lifetime` changed to `-redis.max-connection-age`
    - `-redis.max-idle-conns` removed
    - `-redis.wait-on-pool-exhaustion` removed
+* [CHANGE] TLS configuration for gRPC, HTTP and etcd clients is now marked as experimental. These features are not yet fully baked, and we expect possible small breaking changes in Cortex 1.5. #3198
```
Comment (roidelapluie) [inline CHANGELOG.md:40]:
```suggestion
* [CHANGE] TLS configuration for gRPC, HTTP and etcd clients is now marked as experimental. These features are not yet fully backed, and we expect possible small breaking changes in Cortex 1.5. #3198
```

Comment (pstibrany) [reply CHANGELOG.md:40]: `baked` is correct here, as in "baking the cake". It means they are not finished.

Comment (roidelapluie) [reply CHANGELOG.md:40]: :facepalm: yes sorry. need more :coffee:

**Review by roidelapluie — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (pracucci) [reply CHANGELOG.md:40]: I think `baked` is correct here. It's the sense of "cooked".

**Review by gouthamve — APPROVED** [review]

**Pull request merged**